crates/primitives/src/transaction/eip4844.rs

```diff
@@ -116,3 +150,48 @@ impl TxEip4844 {
         mem::size_of::<u128>() // max_fee_per_data_gas
     }
 }
+
+/// This represents a set of blobs, and its corresponding commitments and proofs.
+#[derive(Clone, Debug, PartialEq, Eq, Default)]
+pub struct BlobTransactionSidecar {
+    /// The blob data.
+    pub blobs: Vec<Blob>,
+    /// The blob commitments.
+    pub commitments: Vec<Bytes48>,
+    /// The blob proofs.
+    pub proofs: Vec<Bytes48>,
+}
+
+impl BlobTransactionSidecar {
+    /// Encodes the inner [BlobTransactionSidecar] fields as RLP bytes, without a RLP header.
+    ///
+    /// This encodes the fields in the following order:
+    /// - `blobs`
+    /// - `commitments`
+    /// - `proofs`
+    pub fn encode_inner(&self, out: &mut dyn bytes::BufMut) {
```

Comment:
should this be pub?

Reply:
yeah this needs to be pub because `eth-wire` needs to use it

Reply:
now is not pub any more because the types are in primitives